public/app/features/dashboard-scene/serialization/DashboardSceneSerializer.ts

```diff
@@ -47,6 +48,12 @@ export interface DashboardSceneSerializerLike<T, M, I = T> {
   getPanelIdForElement: (elementId: string) => number | undefined;
   getElementIdForPanel: (panelId: number) => string | undefined;
   getElementPanelMapping: () => Map<string, number>;
+  getDefaultDsReferencesForElement: (elementId: string) => ElementDsReferences | undefined;
+}
+
+interface ElementDsReferences {
+  queries: { [refId: string]: boolean };
+  variables: { [name: string]: boolean };
```

Comment:
The DS variables are at the dashboard level, why do we have them by element?

I imagined something like:
``` ts
private defaultDSRefs = {
  panelQueries: new Map({ <elementId> : new Set([<refId>, <refId>, ...])}),
  variablesQueries: new Set([<variableName>, <variableName>, ...]),
  annotationsQueries: new Set([<annotationName>, <annotationName>, ...])
}
```
I was thinking that Set could help us check if the variable, panel, or annotation is undefined.
```ts
// Check
this.defaultDSRefs.panels.get(elementId)?.has(refId);

// Remove after DS updated
this.defaultDSRefs.panels.get(elementId)?.remove(refId);

// Initialize
this.defaultDSRefs.panels.get(elementId)?.add(refId);
```
What do you think? 😄